### PR TITLE
Ponyfill useId

### PR DESCRIPTION
**Pull request by Johennes**
Inspired by https://github.com/ezcater/recipe/blob/dc253df74f0b9c509f13172980eb8c1a0defb62a/packages/recipe/src/utils/hooks/useUniqueId.ts.

Fixes: vector-im/compound#224
Replaces: https://github.com/vector-im/compound-web/pull/85

### CODE REVIEW

**Comment by Johennes** [comment]
Percy is now failing because of the quota.

> [percy] Error: This organization has exceeded the limits of the Percy BrowserStack plan. Administrators can upgrade here: https://percy.io/organizations/c8fecada/billing

File: src/utils/useId.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import * as React from "react";
```
Comment (t3chguy) [inline src/utils/useId.ts:1]: copyright pls

**Review by t3chguy — APPROVED** [review]

File: src/utils/useId.ts (hunk 1 above)
Comment (Johennes) [reply src/utils/useId.ts:17]: Ah geez. You can tell I don't get to add new files very often. 🤦‍♂️

Thanks, should be fixed now.

**Review by germain-gg — COMMENTED** [review]
Code looks good to me,

Could we add an ESLint rule to make `useId` a restricted import and suggest using your ponyfill instead? 
I believe we're using https://eslint.org/docs/latest/rules/no-restricted-imports on EW

Does not have to block this pull request, but a quick follow up would be appreciated

**Comment by Johennes** [comment]
Ok, let me see if I can get that going tonight. Will land this in the interim.

**Comment by Johennes** [comment]
Sorry, I didn't get around to do this last night. I have put in https://github.com/vector-im/compound/issues/228 and will try to come back to it.

**Pull request merged**